### PR TITLE
[CARBONDATA-1993] Carbon properties default values fix and corresponding template and document correction.

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1715/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3036/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3045/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1814/

**Comment by kushalsaha** [comment]
retest SDV please

**Comment by kushalsaha** [comment]
retest SDV please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3047/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2014/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3250/

**Comment by mohammadshahidkhan** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3203/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3205/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2029/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3265/

**Comment by mohammadshahidkhan** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3282/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2052/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3357/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2120/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3275/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2152/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3389/

File: conf/carbon.properties.template [hunk 1]
```diff
@@ -17,29 +17,25 @@
 #
 
 #################### System Configuration ##################
-#Mandatory. Carbon Store path
-carbon.storelocation=hdfs://hacluster/Opt/CarbonStore
+#Optional. Carbon Store path
```
Comment (ravipesala) [inline conf/carbon.properties.template]: Mention that if it is not specified it takes spark warehouse path

File: conf/carbon.properties.template [hunk 2]
```diff
@@ -76,22 +72,16 @@ carbon.enable.quick.filter=false
...
-#carbon.load.metadata.lock.retries=3
 ##Minimum blocklets needed for distribution.
 #carbon.blockletdistribution.min.blocklet.size=10
 ##Interval between the retries to get the lock
 #carbon.load.metadata.lock.retry.timeout.sec=5
 ##Temporary store location, By default it will take System.getProperty("java.io.tmpdir")
-#carbon.tempstore.location=/opt/Carbon/TempStoreLoc
-##data loading records count logger
-#carbon.load.log.counter=500000
+#carbon.tempstore.location
```
Comment (ravipesala) [inline conf/carbon.properties.template]: Are we really using this? I think we always depends on eith java tmp dir or get tmp directoris from spark/yarn. Please reverify and remove if not used

File: conf/carbon.properties.template [hunk 3]
```diff
@@ -110,7 +100,7 @@ carbon.enable.quick.filter=false
 ##Percentage to identify whether column cardinality is more than configured percent of total row count
 #high.cardinality.row.count.percentage=80
```
Comment (ravipesala) [inline conf/carbon.properties.template]: This is also not used I guess, please check and remove

File: conf/carbon.properties.template [hunk 4]
```diff
@@ -76,22 +72,16 @@ carbon.enable.quick.filter=false
...
 #carbon.blockletdistribution.min.blocklet.size=10
 ##Interval between the retries to get the lock
 #carbon.load.metadata.lock.retry.timeout.sec=5
 ##Temporary store location, By default it will take System.getProperty("java.io.tmpdir")
-#carbon.tempstore.location=/opt/Carbon/TempStoreLoc
-##data loading records count logger
-#carbon.load.log.counter=500000
+#carbon.tempstore.location
 ##To dissable/enable carbon block distribution
 #carbon.custom.block.distribution=false
```
Comment (ravipesala) [inline conf/carbon.properties.template]: This property is now changed to `carbon.task.distribution` and its default value is `block`

File: docs/configuration-parameters.md [hunk 5]
```diff
@@ -32,10 +32,10 @@ This section provides the details of all the configurations required for the Car
...
-| carbon.badRecords.location | /opt/Carbon/Spark/badrecords | Path where the bad records are stored. |
-| carbon.data.file.version | 3 | If this parameter value is set to 1, then CarbonData will support the data load which is in old format(0.x version). If the value is set to 2(1.x onwards version), then CarbonData will support the data load of new format only. The default value for this parameter is 3(latest version is set as default version). It improves the query performance by ~20% to 50%. For configuring V3 format explicitly, add carbon.data.file.version = V3 in carbon.properties file. |
+| carbon.storelocation |  | Location where CarbonData will create the store, and write the data in its own format. NOTE: Store location should be in HDFS. |
```
Comment (ravipesala) [inline docs/configuration-parameters.md]: Here also mention that if it is not specified it takes spark warehouse path

File: conf/carbon.properties.template (hunk 4 above)
Comment (mohammadshahidkhan) [reply conf/carbon.properties.template]: The property still in use 
 val useCustomDistribution =
          CarbonProperties.getInstance().getProperty(
            CarbonCommonConstants.CARBON_CUSTOM_BLOCK_DISTRIBUTION,
            "false").toBoolean ||
          carbonDistribution.equalsIgnoreCase(CarbonCommonConstants.CARBON_TASK_DISTRIBUTION_CUSTOM)
        if (useCustomDistribution)

File: conf/carbon.properties.template (hunk 3 above)
Comment (mohammadshahidkhan) [reply conf/carbon.properties.template]: not used  removed

File: conf/carbon.properties.template (hunk 2 above)
Comment (mohammadshahidkhan) [reply conf/carbon.properties.template]: We have used this in CarbonAlterTableCompactionCommand, but i think there also we can use java tmp dir. so removed the property and usage also.

File: docs/configuration-parameters.md (hunk 5 above)
Comment (mohammadshahidkhan) [reply docs/configuration-parameters.md]: Added

File: conf/carbon.properties.template (hunk 1 above)
Comment (mohammadshahidkhan) [reply conf/carbon.properties.template]: added

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3463/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2223/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3466/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2226/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3331/

**Comment by mohammadshahidkhan** [comment]
retest SDV please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3334/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3337/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 6]
```diff
@@ -1264,18 +1231,7 @@
...
-   * property for offheap based processing
-   */
-  @CarbonProperty
-  public static final String USE_OFFHEAP_IN_QUERY_PROCSSING = "use.offheap.in.query.processing";
-
-  /**
-   * default value of offheap based processing
-   */
-  public static final String USE_OFFHEAP_IN_QUERY_PROCSSING_DEFAULT = "true";
+  public static final String ENABLE_UNSAFE_IN_QUERY_EXECUTION_DEFAULTVALUE = "true";
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Please make ENABLE_UNSAFE_COLUMN_PAGE_LOADING = "enable.unsafe.columnpage" also "true" by default as its the common configuration for query also.

Comment (mohammadshahidkhan) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Change default  value for "enable.unsafe.columnpage"  to true

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2248/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3492/

**Comment by mohammadshahidkhan** [comment]
retest this please

**Comment by KanakaKumar** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2265/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3503/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3353/

**Pull request closed without merging**